pkgs/build-support/rust/build-rust-package/default.nix

```diff
@@ -15,145 +15,149 @@
+
+  in
+
+  # Tests don't currently work for `no_std`, and all custom sysroots are currently built without `std`.
+    # See https://os.phil-opp.com/testing/ for more information.
+  assert useSysroot -> !(finalAttrs.doCheck or true);
+  {
+    inherit buildAndTestSubdir cargoDeps;
+
+    # buildType is checked for 4 of the cargo built in types
+    # directly specifying cargoBuildType can be used as a escape-hatch
+    cargoBuildType = assert finalAttrs ? buildType -> lib.assertOneOf "buildType" finalAttrs.buildType [ "release" "debug" "test" "bench" ];
+      pickPreviousOrFinalOrDefault "cargoBuildType" "buildType" "release";
+
+    cargoCheckType = pickPreviousOrFinalOrDefault "cargoCheckType" "checkType" finalAttrs.cargoBuildType;
+
+    cargoBuildNoDefaultFeatures = pickPreviousOrFinalOrDefault "cargoBuildNoDefaultFeatures" "buildNoDefaultFeatures" false;
+
+    cargoCheckNoDefaultFeatures = pickPreviousOrFinalOrDefault "cargoCheckNoDefaultFeatures" "checkNoDefaultFeatures"
+      finalAttrs.cargoBuildNoDefaultFeatures;
+
+    cargoBuildFeatures = pickPreviousOrFinalOrDefault "cargoBuildFeatures" "buildFeatures" "";
+
+    cargoCheckFeatures = pickPreviousOrFinalOrDefault "cargoCheckFeatures" "checkFeatures" finalAttrs.cargoBuildFeatures;
+
+    patchRegistryDeps = ./patch-registry-deps;
+
+    nativeBuildInputs = maybeSetListFromPrevious "nativeBuildInputs" ++ [
+      cacert
+      git
+      cargoBuildHook
+      cargoCheckHook
+      cargoInstallHook
+      cargoSetupHook
+      rustc
+    ];
+
+    buildInputs = maybeSetListFromPrevious "buildInputs"
     ++ lib.optionals stdenv.hostPlatform.isDarwin [ libiconv ]
     ++ lib.optionals stdenv.hostPlatform.isMinGW [ windows.pthreads ];
 
-  patches = cargoPatches ++ patches;
+    # commented for now to keep drv hash from changing
+    #cargoPatches = maybeSetListFromPrevious "cargoPatches";
+    #patches = finalAttrs.cargoPatches ++ maybeSetListFromPrevious "patches";
+
+    patches = (finalAttrs.cargoPatches or [ ]) ++ maybeSetListFromPrevious "patches";
+
+    PKG_CONFIG_ALLOW_CROSS = previousAttrs.PKG_CONFIG_ALLOW_CROSS or
+      (if stdenv.buildPlatform != stdenv.hostPlatform then 1 else 0);
 
-  PKG_CONFIG_ALLOW_CROSS =
-    if stdenv.buildPlatform != stdenv.hostPlatform then 1 else 0;
+    postUnpack = ''
+      eval "$cargoDepsHook"
 
-  postUnpack = ''
-    eval "$cargoDepsHook"
+      export RUST_LOG=${finalAttrs.logLevel or ""}
+    '' + maybeSetStr "postUnpack";
 
-    export RUST_LOG=${logLevel}
-  '' + (args.postUnpack or "");
+    configurePhase = previousAttrs.configurePhase or ''
+      runHook preConfigure
+      runHook postConfigure
+    '';
 
-  configurePhase = args.configurePhase or ''
-    runHook preConfigure
-    runHook postConfigure
-  '';
+    doCheck = previousAttrs.doCheck or true;
 
-  doCheck = args.doCheck or true;
+    strictDeps = previousAttrs.strictDeps or true;
```

Comment:
same here, previously we'd always set this to `true`. probably better to leave it like that anyway.